### PR TITLE
Fix logic that prevents logging duplicate errors

### PR DESCRIPTION
**Pull request by olegrise**
## Description

The logic around `fileInError` was totally broken. 
- We expect the `details` to have `params` field [here](https://github.com/Rise-Vision/rise-common-component/compare/stage/fix-duplicate-error?expand=1#diff-a68527ff1ca3585485c66497e290c037ad0113f352ca9c4e10152ecaf70ced5cL43) , but it only has 'filePath, errorMessage, errorDetail` see [here](https://github.com/Rise-Vision/rise-common-component/blob/3da05a73a2672378110f8c03f7f9d70ec4713ac8/src/watch-files-mixin.js#L148)
- When comparing objects, we expect `fileInError` to have `details` field [here](https://github.com/Rise-Vision/rise-common-component/compare/stage/fix-duplicate-error?expand=1#diff-a68527ff1ca3585485c66497e290c037ad0113f352ca9c4e10152ecaf70ced5cL176), but it does not exist.

I decided to keep the `JSON.stringify` objects comparison. It's not ideal, but considering that we contract the `details` objects ourselves [here](https://github.com/Rise-Vision/rise-common-component/blob/3da05a73a2672378110f8c03f7f9d70ec4713ac8/src/watch-files-mixin.js#L148) and then save it to the list without modification, I think it's okay to use `JSON.stringify`.

One thing to mention is that in-spite of this fix I still see the duplicate errors being logged. There are two reasons for that:
1) **The main reason is that content sentinel sends "FILE-ERROR" first and that results in the file being added to the `_managedFilesInError`, but then, all of a sudden, it sends "FILE-UPDATE" with status "STALE" which causes file to be removed from the `_managedFilesInError`. On the next schedule cycle the "FILE-ERROR" is sent again and error is logged again because `_managedFilesInError` is now empty.**
2) Each instance of the `rise-image` has its own `_managedFilesInError` list.

## Motivation and Context
Prevents logging duplicate errors. Partial fix for https://github.com/Rise-Vision/rise-image/issues/102

## How Has This Been Tested?
Tested with rise-image that was build referencing this change. Then used proxy to load local version of the image.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

**Review by santiagonoguez — APPROVED** [review]
@olegrise  yes, much simpler.

Your initial comments are also very helpful regarding the behavior I'm seeing on content sentinel side. Thanks !

**Comment by olegrise** [comment]
@stulees @santiagonoguez I don't know how well content sentinel handles this, but I see that [startWatch](https://github.com/Rise-Vision/rise-common-component/blob/8e84c85824421fe41df2817327a7c4aedc67ddcf/src/watch-files-mixin.js#L110) is being called by rise-image again and again for the same files. Basically rise-image calls `startWatch` every time it receives `start` event. I wonder if that behaviour contributes to the issue.

**Comment by stulees** [comment]
@olegrise Given this has been functionality running in production for some time when RLS is used, have you validated with rise-image running on player and forcing the error scenario?

**Comment by stulees** [comment]
@olegrise Re: start watch, there is a flag `watchInitiated` that prevents facilitating the sending of a watch message whenever this function is called if it has already been initiated. https://github.com/Rise-Vision/rise-common-component/blob/master/src/watch-files-mixin.js#L118

**Comment by olegrise** [comment]
@stulees I haven't validated with Player. Will do it now.

Re: start watch - sorry, I wasn't clear. The `startWatch` is called and then [watchFn](https://github.com/Rise-Vision/rise-common-component/blob/8e84c85824421fe41df2817327a7c4aedc67ddcf/src/watch-files-mixin.js#L127) is also called. It seems that `watchInitiated` does nothing. I will take another look to see why `watchInitiated` is not doing anything.

**Comment by olegrise** [comment]
@stulees re `watchInitiated` flag - here is what happens:
- rise-image receives `rise-presentation-play` and calls `_reset()` function https://github.com/Rise-Vision/rise-image/blob/master/src/rise-image.js#L117
- `reset()` calls `_stop()` https://github.com/Rise-Vision/rise-image/blob/master/src/rise-image.js#L190
- `_stop()` calls `super.stopWatch()` https://github.com/Rise-Vision/rise-image/blob/ee9789ddc48f29c6b6bb517dc419ed36ccec08e1/src/rise-image.js#L404
- `stopWatch()` resets `_watchInitiated` flag https://github.com/Rise-Vision/rise-common-component/blob/8e84c85824421fe41df2817327a7c4aedc67ddcf/src/watch-files-mixin.js#L139

To recap, on every `rise-presentation-play` rise-image re-adds files to the watch.

**Comment by stulees** [comment]
@olegrise That flow makes sense, thanks for recapping. IMO, I don't think this is incorrect behaviour on the part of the component. It also should not have an adverse effect on Content Sentinel where a component is "reset" and then sends a WATCH for files again. Content Sentinel responsibility in this scenario is inform the consumer on the status. Components have behaved this way with RLS, it should be no different with Content Sentinel.

**Comment by olegrise** [comment]
@stulees I don't disagree. From the content sentinel point if view it should not matter how many times a consumer adds a file to the watch. On other hand this something that we can potentially optimize in rise-image when we have time.

**Comment by olegrise** [comment]
@stulees forced the error in the Chrome OS player.  Works like a charm! No surprises.

**Review by stulees — APPROVED** [review]

**Comment by stulees** [comment]
@olegrise Thanks for validating that.
